core/src/Kokkos_Atomic.hpp

```diff
@@ -333,4 +339,5 @@ extern KOKKOS_INLINE_FUNCTION void unlock_address_rocm_space(void* ptr);
 
 //----------------------------------------------------------------------------
 
+#endif /* !KOKKOS_IMPL_ENABLE_DESUL_ATOMICS */
```

Comment:
Given how big this header is, does it make more sense to do something like:
```cpp
#ifdef KOKKOS_IMPL_ENABLE_DESUL_ATOMICS
// put desul_stuff here
#else
#include <KokkosNoDesulAtomics.hpp>
#endif
```
I mainly ask because the `#endif /* !KOKKOS_IMPL_ENABLE_DESUL_ATOMICS */`
is several hundred lines away from the `#ifdef`.

Reply:
Also is this a place we should consider inline namespaces?

Reply:
eventually (i.e. when everything is testing and working, all the other shit will be deleted) There will not be a public option to switch back and forth. I imagine we will leave a non-public option in there for one release in case we run into trouble downstream and need to see if the atomics are the issue, but otherwise this will all just go away .